tests/core/gui/test_password_change.py

```diff
@@ -149,3 +149,24 @@ def _create_change_password_dialog():
 
     async with aqtbot.wait_signals([lw.login_with_password_clicked, tabw.logged_in]):
         await aqtbot.mouse_click(lw.button_login, QtCore.Qt.LeftButton)
+
+
+@pytest.mark.gui
+@pytest.mark.trio
+async def test_change_password_canceled(
+    aqtbot, running_backend, logged_gui, qt_thread_gateway, autoclose_dialog
+):
+    d_w = logged_gui.test_get_devices_widget()
+
+    assert d_w is not None
+    async with aqtbot.wait_signal(d_w.list_success):
```

Comment:
instead of waiting on signals which is cumbersome, not reliable and error prone (rumors say it also gives cancer... :trollface: ) it's better to use `aqtbot.wait_until(callable)`:

```python
def _devices_displayed():
        assert d_w.isVisible()
        assert d_w.layout_devices.count() == 2

await aqtbot.wait_until(_devices_displayed)
```

(btw pay attention to use `widget.isVisible()` and not `not widget.isHidden()`, the two are not related)

Reply:
fixed, thanks for the info